src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/PersistedObjectHandler.php

```diff
@@ -97,6 +97,14 @@ public function createEntity(
         }
         
         $retrievedEntity = DataObjectHandler::getInstance()->getObject($entity);
+
+        if ($retrievedEntity === null) {
+            throw new TestReferenceException(
+                "Entity \"" . $entity . "\" does not exist." .
+                "\nException occurred parsing action at StepKey \"" . $key . "\""
```

Comment:
Missed this in the first review. Should we say "Exception occurred executing action at StepKey..."? "Parsing" to me sounds more like during generation.